### DISCUSSION TITLE
Tasmota Nous A1 / problem with TLS secured MQTT

### DISCUSSION THREAD

**Discussion by Pauline811** [arendst/Tasmota#18760] [Q&A]
i've tried to connect my nous a1 but i always get error code r-4. i noticed as well that i should have the opportunity to enable via check box MQTT TLS, but i dont  have this option.  Does anyone know what to enable TLS and connect to the MQTT broker?

**Comment by sfromis** [comment]
You need a build including the optional feature of TLS, and correct config of the details. https://tasmota.github.io/docs/TLS/

**Reply by Pauline811** [reply]
looks like its only working for esp32 and not esp82 or do i get that wrong?

**Reply by sfromis** [reply]
MQTT-TLS does also work for ESP8266, with a suitable build and config.
